connect/api/src/main/java/org/apache/kafka/connect/sink/ErrantRecordReporter.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.connect.sink;
+
+import java.util.concurrent.Future;
+
+public interface ErrantRecordReporter {
+
+
+  /**
+   * Report a problematic record and the corresponding error to be written to the sink
+   * connector's dead letter queue (DLQ).
```

Comment:
this is separate from the Connect DLQ - maybe we should avoid mentioning it in the docs to make sure its a different concept? 

```suggestion
   * Report a problematic record and the corresponding error to be written to the sink
   * connector's error topic.
```

or something similar

Reply:
It is now getting written to the same DLQ, so I am just going to keep it like this.